docs/docs/plugin.md

```diff
@@ -0,0 +1,166 @@
+  The Marathon plugin functionality is considered alpha, so use this feature at you own risk. We might add, change, or delete any functionality described in this document.  
+</div>
+
+## Overview
+
+<p class="text-center">
+  <img src="{{ site.baseurl}}/img/plugin-mechanism.png" width="552" height="417" alt="">
+</p>
+
+
+Starting with Marathon 0.12, we allow you to extend Marathon functionality with plugins.
+The plugin mechanism has the following components:
+
+- __Extension-Aware Functionality:__ Marathon makes only specific functionality available for customization through hooks implemented into the main system. These hooks can be refined and extended with external plugins.
+- __Extension Interface:__ In order to extend functionality with a plugin, you must implement the extension interface. The extension interface is defined as a Scala trait or Java interface.
+- __Plugin Extension:__ A plugin extension is the implementation of a defined extension interface. The plugin needs to be compiled and packaged into one or more separate jar files. 
+- __Plugin Descriptor:__ The descriptor defines which set of plugins should be enabled in one instance of Marathon.
+  
+
+
+## Plugin Interface
+
+A separate Marathon plugin interface jar is published in the following Maven repository with every Marathon release:
+
+```
+http://downloads.mesosphere.io/maven
+```
+
+with following dependency:
+
+SBT:
+
+```
+"mesosphere.marathon" %% "plugin-interface" % "x.x.x" % "provided"
+
+```
+
+Maven:
+
+```
+<dependency>
+    <groupId>mesosphere.marathon</groupId>
+    <artifactId>plugin-interface_2.11</artifactId>
+    <version>x.x.x</version>
+</dependency>
+```
+
+Be sure to always use the same version of the plugin interface as the version of Marathon in production.
+
+
+
+## Extension Functionality
+
+To create an extension, you must implement an extension interface from the plugin interface method and put it in a location available to Marathon so that it can be loaded by the ServiceLoader.
+
+### ServiceLoader
+
+Marathon uses the [ServiceLoader](https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html) 
+to make a service provider available to a running Marathon instance. For background information on that topic, please see [Creating Extensible Applications](https://docs.oracle.com/javase/tutorial/ext/basics/spi.html).
+
+### Dependencies on other libraries
+
+The writer of the plugin can define dependencies on other libraries and use those dependencies if following requirements are met:
+
+- Marathon already has a dependency to that library. You can see all Marathon dependencies by checking out Marathon and running `sbt dependencyTree`. Such a dependency should be defined as `provided` in your build tool of choice.
+  _Please make sure, you use the exact same version of that library!_
+- The library does not conflict with any library bundled with Marathon (for instance, different byte code manipulation libraries are known to interfere).
```

Comment:
Do both of the things on this list need to be true in order to define dependencies to libraries? Or are the "provided" and "compile" dependencies different?